### PR TITLE
PIMS-2079

### PR DESCRIPTION
**Pull request by jdtoombs**
Very small PR - tests that the project number is generated when creating a new project.

Had a number of other small tests written that ended up being covered by Jeremy's last PR. Finally getting more comfortable with the back end side of things though!

### CODE REVIEW

File: backend/tests/unit/dal/Services/ProjectServiceTest.cs [hunk 1]
```diff
@@ -367,6 +367,27 @@ public void Get_Sensitive_AsAdmin()
...
+            var helper = new TestHelper();
+            var user = PrincipalHelper.CreateForPermission(Permissions.PropertyAdd).AddAgency(1);
+
+            var dbName = StringHelper.Generate(10);
+            using var init = helper.InitializeDatabase(dbName, user);
+            var project = init.CreateProject(1);
+            init.SaveChanges();
+
+            // Act 
+
```
Comment (Fosol) [inline backend/tests/unit/dal/Services/ProjectServiceTest.cs:384]: Your Act is missing.  You should be calling the `service.Add(...)`
You are currently testing our test code, not the DAL with this.

File: backend/tests/unit/dal/Services/ProjectServiceTest.cs [hunk 2]
```diff
@@ -389,7 +390,31 @@ public void Get_Sensitive_AsAdmin()
...
+        #region Add 
+        [Fact]
+        public void Add_Project()
+        {
+            // Arrange
+            var helper = new TestHelper();
+            var user = PrincipalHelper.CreateForPermission(Permissions.PropertyAdd).AddAgency(1);
+            var project = EntityHelper.CreateProject(1);
+
+            var options = Options.Create(new ProjectOptions(){NumberFormat="TEST-{0:00000}"});
```
Comment (Fosol) [inline backend/tests/unit/dal/Services/ProjectServiceTest.cs:405]: Nice!

File: backend/tests/unit/dal/Services/ProjectServiceTest.cs [hunk 3]
```diff
@@ -389,7 +390,31 @@ public void Get_Sensitive_AsAdmin()
...
+        /// User with appropriate permission successfully adds new project. Project Number is auto-generated.
+        /// </summary>
+        #region Add 
+        [Fact]
+        public void Add_Project()
+        {
+            // Arrange
+            var helper = new TestHelper();
+            var user = PrincipalHelper.CreateForPermission(Permissions.PropertyAdd).AddAgency(1);
+            var project = EntityHelper.CreateProject(1);
```
Comment (Fosol) [inline backend/tests/unit/dal/Services/ProjectServiceTest.cs:403]: Now you may want to prove that the number is generated by starting with a different number here.

Comment (jdtoombs) [reply backend/tests/unit/dal/Services/ProjectServiceTest.cs:403]: Will add that in before I merge!

**Pull request merged**